app/views/contexts/section_summary_context.py

```diff
@@ -237,3 +328,159 @@ def _get_safe_page_title(self, title):
+                    if (
+                        answer.answer_id in related_answers
+                        and answer.list_item_id == list_id
+                    ):
+                        edit_block_id = self._schema.get_edit_block_for_list_collector(
+                            list_collector_block_id
+                        )["id"]
+
+                        question = dict(
+                            self._schema.get_add_block_for_list_collector(  # type: ignore
+                                list_collector_block_id
+                            ).get(
+                                "question"
+                            )
+                        )
+
+                        question["answers"] = list(question["answers"])[1:]
+
+                        block_schema = {
+                            "id": edit_block_id,
+                            "title": None,
+                            "number": None,
+                            "type": "ListCollector",
+                            "for_list": current_list.name,
+                            "question": question,
+                        }
+                        block = [
+                            Block(
+                                block_schema,
+                                answer_store=self._answer_store,
+                                list_store=self._list_store,
+                                metadata=self._metadata,
+                                response_metadata=self._response_metadata,
+                                schema=self._schema,
+                                location=Location(
+                                    list_name=current_list.name,
+                                    list_item_id=list_id,
+                                    section_id=self.section["id"],
+                                ),
+                                return_to="section-summary",
+                                return_to_block_id=None,
+                            ).serialize()
+                        ]
+
+                        related_answers_dict[list_id] = block
+
+            return related_answers_dict
+
+    def _get_answer_title(self, list_collector_block: Mapping[str, Any]) -> str:
+        if list_collector_block["add_block"].get("question_variants"):
+            variant_label = choose_variant(
+                list_collector_block["add_block"],
+                self._schema,
+                self._metadata,
+                self._response_metadata,
+                self._answer_store,
+                self._list_store,
+                variants_key="question_variants",
+                single_key="question",
+                current_location=self.current_location,
+            )["answers"][0]["label"]
+
+            return variant_label
+
+        return list_collector_block["add_block"]["question"]["answers"][0]["label"]
+
+    def _get_answer_id(self, list_collector_block: Mapping[str, Any]) -> str:
```

Comment:
As per my comments above, this should be dropped in favour of an explicit id.

Reply:
Ok, note for myself: this will be resolved after adding `item_anchor_answer_id` (one of the comments above).

Reply:
Explicit new id added now, thanks.